src/models/Room.ts

```diff
@@ -0,0 +1,135 @@
+import { PowerLevelsEventContent } from "./events/PowerLevelsEvent";
+
+export enum RoomPreset {
+	/**
+	 * Sets:
+	 * - join_rules to `invite`
+	 * - history_visibility to `shared` 
+	 * - guest_access to `can_join`
+	 */
+	PrivateChat = "private_chat",
+	/**
+	 * Sets:
+	 * - join_rules to `invite`
+	 * - history_visibility to `shared` 
+	 * - guest_access to `can_join`
+	 * - All invitees are given the same power level as the room creator.
+	 */
+	TrustedPrivateChat = "trusted_private_chat",
+	/**
+	 * Sets:
+	 * - join_rules to `public`
+	 * - history_visibility to `shared` 
+	 * - guest_access to `forbidden`
+	 */
+	PublicChat = "public_chat",
+}
+
+export enum RoomVisibility {
+	/**
+	 * This visibility indicates that the room will be shown in the published room list. 
+	 */
+	Public = "public",
+	/**
+	 * This visibility indicates that the room will not be included in published room list.
+	 */
+	Private = "private",
+}
+
+export interface RoomCreateOptions {
+	/**
+	 * Extra keys, such as m.federate, to be added to the content of the m.room.create event.
+	 * The server will overwrite the following keys: `creator`, `room_version`.
+	 * Future versions of the specification may allow the server to overwrite other keys.
+	 */
+	creation_content?: {
+		"m.federate"?: boolean,
+		[key: string]: unknown
```

Comment:
trailing comma, also please describe it as a `type` for `Record<string, unknown> & { "m.federate"?: boolean }`

Reply:
Done, but also fyi: 

![image](https://user-images.githubusercontent.com/2072976/177304680-fd7cb0a6-a077-4461-bef0-956293960771.png)

Reply:
ITYM semicolons, because other parts of the code are using them and `yarn lint` fails unless I semicolon'd myself. If that's not right, I might just ask you to do the tidy up rather than another round trip.